rcl/src/rcl/logging_rosout.c

```diff
@@ -173,6 +173,8 @@ rcl_ret_t rcl_logging_rosout_init_publisher_for_node(
   const rosidl_message_type_support_t * type_support =
     rosidl_typesupport_c__get_message_type_support_handle__rcl_interfaces__msg__Log();
   rcl_publisher_options_t options = rcl_publisher_get_default_options();
+  options.qos.depth = 1000;
```

Comment:
why 1000? why don't just using KEEP_ALL?

Reply:
Since that might result in infinite memory usage? The goal is to cover the race to allow subscribers coming up shortly after a node starts publishing rosout messages to receive - not a for alate joiner an hour later to get all rosout message from the beginning.

Reply:
I actually agree with your previous comment https://github.com/ros2/rcl/issues/581#issuecomment-591519046, exposing this to the user would be ideal and likely the best solution. But this isn't an unreasonable default.

Reply:
> Since that might result in infinite memory usage?

That's not how `KEEP_ALL` works exactly. There's a DDS setting we don't currently expose, `MAX_SAMPLES` and `MAX_SAMPLES_PER_INSTANCE` which determines when `KEEP_ALL` stops storing things in the history cache.

Reply:
How does `KEEP_ALL` currently behave then without these additional options exposed or set to custom values?